File: WordTracker.py
class WordTracker:
    def __init__(self, sentence):
        self.sentence = sentence
        self.word_counts = {}

    def count_words(self):
        words = self.sentence.split(" ")
        for word in words:
            if word in self.word_counts:
                self.word_counts[word] += 1
            else:
                self.word_counts[word] = 1

    def get_word_frequency(self, word):
        return self.word_counts.get(word, 0)

    def print_frequencies(self):
        for word, count in self.word_counts.items():
            print(f"{word}: {count}")

tracker = WordTracker("hello world hello again world")
tracker.count_words()
tracker.print_frequencies()
print("Frequency of 'hello':", tracker.get_word_frequency("hello"))

File: WordTracker.java
import java.util.HashMap;
import java.util.Map;

public class WordTracker {
    public String sentence;
    public Map<String, Integer> wordCounts;

    public WordTracker(String sentence) {
        this.sentence = sentence;
        this.wordCounts = new HashMap<>();
    }

    public void countWords() {
        String[] words = sentence.split(" ");
        for (String word : words) {
            if (wordCounts.containsKey(word)) {
                wordCounts.put(word, wordCounts.get(word) + 1);
            } else {
                wordCounts.put(word, 1);
            }
        }
    }

    public int getWordFrequency(String word) {
        return wordCounts.getOrDefault(word, 0);
    }

    public void printFrequencies() {
        for (String word : wordCounts.keySet()) {
            System.out.println(word + ": " + wordCounts.get(word));
        }
    }

    public static void main(String[] args) {
        WordTracker tracker = new WordTracker("hello world hello again world");
        tracker.countWords();
        tracker.printFrequencies();
        System.out.println("Frequency of 'hello': " + tracker.getWordFrequency("hello"));
    }
}
